From: Boyuan Yang <byang@debian.org>
Date: Fri, 15 Aug 2025 20:23:36 -0400
Subject: CMakeLists.txt: Do not enable NEON for armel armhf

According to https://wiki.debian.org/ArchitectureSpecificsMemo#armel
the armhf architecture does not guarantee NEON. Do not enable NEON
on armhf. Also disable NEON for armel to prevent FTBFS.

Note that after 2025-12-15 the upstream decided that non-NEON arm
builds are no longer supported. This patch is provided as-is and
may or may not work. See https://issues.chromium.org/issues/469092869
as well as
https://chromium.googlesource.com/libyuv/libyuv/+/821b9c5de1cf232f51c26e1eaad4689fc2317949%5E%21/
for more details.

Forwarded: not-needed
---
 CMakeLists.txt | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9db2aaa..68c5e9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,16 +90,9 @@ endif()
 if(NOT MSVC)
   string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" arch_lowercase)
 
+# Debian-specific: Do not enable Arm Neon kernels on non-arm64 architectures.
   if(arch_lowercase MATCHES "^arm" AND NOT arch_lowercase STREQUAL "arm64")
-    # Enable Arm Neon kernels.
-    add_definitions(-DLIBYUV_NEON=1)
-    add_library(${ly_lib_name}_neon OBJECT
-      ${ly_src_dir}/compare_neon.cc
-      ${ly_src_dir}/rotate_neon.cc
-      ${ly_src_dir}/row_neon.cc
-      ${ly_src_dir}/scale_neon.cc)
-    target_compile_options(${ly_lib_name}_neon PRIVATE -mfpu=neon)
-    list(APPEND ly_lib_parts $<TARGET_OBJECTS:${ly_lib_name}_neon>)
+    message("Debian-specific: Not enabling NEON on ${arch_lowercase}.")
   endif()
 
   if(arch_lowercase STREQUAL "aarch64" OR arch_lowercase STREQUAL "arm64")
